src/HotChocolate/Core/test/Types.Tests/Types/Relay/PolymorphicGlobalIdsTypeInterceptor.cs

```diff
@@ -0,0 +1,160 @@
+                                argumentDefinition,
+                                idInfo.Value.NodeTypeName,
+                                idInfo.Value.IdRuntimeType);
+                        }
+                    }
+                }
+            }
+
+            base.OnBeforeCompleteType(completionContext, definition, contextData);
+        }
+
+        private static void InsertFormatter(
+            ITypeCompletionContext completionContext,
+            ArgumentDefinition definition,
+            NameString typeName,
+            Type idRuntimeType)
+        {
+            var formatter = new PolymorphicGlobalIdInputValueFormatter(
+                typeName,
+                idRuntimeType,
+                GetIdSerializer(completionContext));
+
+            IInputValueFormatter? defaultFormatter = definition.Formatters
+                .FirstOrDefault(f => f is GlobalIdInputValueFormatter);
+
+            if (defaultFormatter == null)
+            {
+                definition.Formatters.Add(formatter);
+            }
+            else
+            {
+                definition.Formatters.Insert(
+                    definition.Formatters.IndexOf(defaultFormatter) - 1,
+                    formatter);
+            }
+        }
+
+        private static (string NodeTypeName, Type IdRuntimeType)? GetIdInfo(
+            ITypeCompletionContext completionContext,
+            ArgumentDefinition definition)
+        {
+            ITypeInspector typeInspector = completionContext.TypeInspector;
+            IDAttribute? idAttribute;
+            IExtendedType? idType;
+
+            if (definition is InputFieldDefinition inputField)
+            {
+                idAttribute = (IDAttribute?)inputField.Property
+                   .GetCustomAttributes(inherit: true)
+                   .SingleOrDefault(a => a is IDAttribute);
+                if (idAttribute == null)
+                {
+                    return null;
+                }
+
+                idType = typeInspector.GetReturnType(inputField.Property, true);
+                
+            }
+            else if (definition.Parameter is not null)
+            {
+                idAttribute = (IDAttribute?)definition.Parameter
+                           .GetCustomAttributes(inherit: true)
+                           .SingleOrDefault(a => a is IDAttribute);
+                if (idAttribute == null)
+                {
+                    return null;
+                }
+
+                idType = typeInspector.GetArgumentType(definition.Parameter, true);
+                
+            }
+            else if (definition.Type is ExtendedTypeReference typeReference)
+            {
+                if (typeReference.Type.Kind == ExtendedTypeKind.Schema)
+                {
+                    return null;
+                }
+
+                throw new NotImplementedException("TODO: Equivalent for IDAttribute here?");
+                idType = typeReference.Type;
+            }
```

Comment:
I wasn't sure what to do in this case, maybe it can just be rolled in to the `else`. It doesn't seem to be needed in my test cases.

```suggestion
```